pkg/promlib/resource/resource.go

```diff
@@ -84,3 +89,110 @@ func (r *Resource) DetectVersion(ctx context.Context, req *backend.CallResourceR
 
 	return r.Execute(ctx, newReq)
 }
+
+func getSelectors(expr string) ([]string, error) {
+	parsed, err := parser.ParseExpr(expr)
+	if err != nil {
+		return nil, err
+	}
+
+	selectors := make([]string, 0)
+
+	parser.Inspect(parsed, func(node parser.Node, nodes []parser.Node) error {
+		switch v := node.(type) {
+		case *parser.VectorSelector:
+			for _, matcher := range v.LabelMatchers {
+				if matcher == nil {
+					continue
+				}
+				if matcher.Name == "__name__" {
+					selectors = append(selectors, matcher.Value)
+				}
+			}
+		}
+		return nil
+	})
+
+	return selectors, nil
+}
+
+// SuggestionRequest is the request body for the GetSuggestions resource.
+type SuggestionRequest struct {
+	// LabelName, if provided, will result in label values being returned for the given label name.
+	LabelName string `json:"labelName"`
+
+	Queries []string `json:"queries"`
+
+	Scopes       []models.ScopeFilter `json:"scopes"`
+	AdhocFilters []models.ScopeFilter `json:"adhocFilters"`
+
+	// Start and End are proxied directly to the prometheus endpoint (which is rfc3339 | unix_timestamp)
+	Start string `json:"start"`
+	End   string `json:"end"`
+
+	// Limit is the maximum number of suggestions to return and is proxied directly to the prometheus endpoint.
+	Limit int64 `json:"limit"`
```

Comment:
Frontend is not sending this parameter.

Reply:
Added the `limit` parameter to `fetchSuggestions` but we're not sending it from `getTagKeys` nor `getTagValues` as neither the original requests were adding them.